macro-pdfviewer-api/src/main/resources/templates/pdfviewer/pdfviewer.vm

```diff
@@ -19,11 +19,19 @@
 ## ---------------------------------------------------------------------------
 ##!source.syntax=xwiki/2.1
 {{velocity output="false"}}
-#macro (maybeGetDocumentAsAuthor $documentReference $asAuthor)
+#macro (areAuthorRightsNeeded $documentReference)
   #if (!$services.security.authorization.hasAccess('view', $documentReference) && $asAuthor &&
       $xwiki.hasProgrammingRights() && $xwiki.exists('XWiki.PDFViewerService'))
-    #set ($document = $xwiki.getDocumentAsAuthor($documentReference))
     #set ($authorRightsNeeded = true)
+  #else
+    #set ($authorRightsNeeded = false)
+  #end
+#end
+
+#macro (maybeGetDocumentAsAuthor $documentReference $asAuthor)
+  #areAuthorRightsNeeded($documentReference)
```

Comment:
You should also pass the `asAuthor` parameter here since you passed it to the `maybeGetDocumentAsAuthor` function. Even if you didnt pass it anywhere, if would still work. But I believe it is better this way for readability purposes.